File: src/Recursion/generateSubset.java
package Recursion;

public class generateSubset {
    static int generateSubset(int []arr,int n,int sum){
        if(n==0)
        {
            if(sum==0)
                return 1;
            else
                return 0;

        }
        return generateSubset(arr,n-1,sum)+generateSubset(arr,n-1,sum-arr[n-1]);
    }
    public static void main(String[] args) {
        int []arr={1,2,3,4,5};
        int n=arr.length;
        int sum=10;
        System.out.println(generateSubset(arr,n,sum));
    }
}

File: src/recursionCpp/generateSubset.cpp
#include <bits/stdc++.h>
using namespace std;

int subset(int arr[],int n,int sum){
    if(n==0){
       return (sum==0)?1:0;
    }
    return subset(arr,n-1,sum)+subset(arr,n-1,sum-arr[n-1]);
}
int main(){
    int arr[]={1,2,3,4,5};
    int n=sizeof(arr)/sizeof(arr[0]);
    int sum=10;
    cout<<subset(arr,n,sum);
    return 0;
}